dartagnan/src/main/java/com/dat3m/dartagnan/expression/op/IOpUn.java

```diff
@@ -94,36 +96,37 @@ public Formula encode(Formula e, SolverContext ctx) {
 			BitvectorFormula bv = (BitvectorFormula)e;
 	    	switch(this) {
     		case MINUS:
-    				return bvmgr.subtract(bvmgr.makeBitvector(32, BigInteger.ZERO), bv);
+    			return bvmgr.subtract(bvmgr.makeBitvector(32, BigInteger.ZERO), bv);
```

Comment:
There is a `bvmgr.negate` function that does what you want (not to be confused with `bvmgr.not` which is the logical ~ operator). It will also keep the precision of the original bv-expression.